src/omnicore/omnicore.cpp

```diff
@@ -3048,6 +2888,54 @@ int mastercore::ClassAgnosticWalletTXBuilder(const std::string& senderAddress, c
     }
 }
 
+void CMPTxList::LoadActivations(int blockHeight)
+{
+    if (!pdb) return;
+
+    Slice skey, svalue;
+    Iterator* it = NewIterator();
+
+    PrintToLog("Loading feature activations from levelDB\n");
+
+    for(it->SeekToFirst(); it->Valid(); it->Next()) {
+       skey = it->key();
+       svalue = it->value();
+       string itData = svalue.ToString();
+
+       std::vector<std::string> vstr;
+       boost::split(vstr, itData, boost::is_any_of(":"), token_compress_on);
+       if (4 != vstr.size()) continue;
+       if (atoi(vstr[2]) != OMNICORE_MESSAGE_TYPE_ACTIVATION || atoi(vstr[0]) != 1) continue; // we only care about valid activations
+
+       uint256 hash(skey.ToString());
+       uint256 blockHash = 0;
+       CTransaction wtx;
+       CMPTransaction mp_obj;
+
+       if (!GetTransaction(hash, wtx, blockHash, true)) {
+           PrintToLog("ERROR: While loading activation transaction %s: tx in levelDB but does not exist.\n", hash.GetHex());
+           continue;
+       }
+       if (0 != ParseTransaction(wtx, blockHeight, 0, mp_obj)) {
```

Comment:
Hehe sorry.

Say, the first time the transaction is processed (and stored in the DB) is block 370000.

If it's at some point later retrieved, say at block 380000, and `nWaterlineBlock` is used, instead of the actual height 370000, then this can have an impact on the parsing rules (because `370000 != 380000`).

See also my other comment. Is this more clear? :) If not, I'll try again hehe.